engine/classes/Elgg/Ajax/Service.php

```diff
@@ -261,4 +267,32 @@ public function appendDeps($hook, $type, $response, $params) {
 		return $response;
 	}
 
-}
+	/**
+	 * Register a view to be available for ajax calls
+	 *
+	 * @param string $view The view name
+	 * @return void
+	 */
+	public function registerView($view) {
+		$this->allowed_views[$view] = true;
+	}
+
+	/**
+	 * Unregister a view for ajax calls
+	 *
+	 * @param string $view The view name
+	 * @return void
+	 */
+	public function unregisterView($view) {
+		unset($this->allowed_views[$view]);
+	}
+
+	/**
+	 * Returns an array of views allowed for ajax calls
+	 * @return string[]
+	 */
+	public function getViews() {
```

Comment:
Plan for later: convert this to isRegisteredView($view) since that's all we need it for.